src/Symfony/Component/Translation/Command/XliffUpdateSourcesCommand.php

```diff
@@ -0,0 +1,203 @@
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Translation\MessageCatalogue;
+use Symfony\Component\Translation\MetadataAwareInterface;
+use Symfony\Component\Translation\Reader\TranslationReaderInterface;
+use Symfony\Component\Translation\Writer\TranslationWriterInterface;
+
+/**
+ * @author Nicolas Rigaud <squrious@protonmail.com>
+ */
+#[AsCommand(name: 'translation:update-xliff-sources', description: 'Update source tags with default locale targets in XLIFF files.')]
+class XliffUpdateSourcesCommand extends Command
+{
+    use TranslationTrait;
+
+    private const FORMATS = [
+        'xlf12' => ['xlf', '1.2'],
+        'xlf20' => ['xlf', '2.0'],
+    ];
+
+    public function __construct(
+        private readonly TranslationWriterInterface $writer,
+        private readonly TranslationReaderInterface $reader,
+        private readonly string $defaultLocale,
+        private readonly array $transPaths = [],
+        private readonly array $enabledLocales = [],
+    ) {
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this
+            ->setDefinition([
+                new InputArgument('paths', InputArgument::OPTIONAL | InputArgument::IS_ARRAY, 'Paths to look for translations.'),
+                new InputOption('format', null, InputOption::VALUE_REQUIRED, 'Override the default output format.', 'xlf12'),
+                new InputOption('domains', null, InputOption::VALUE_OPTIONAL | InputOption::VALUE_IS_ARRAY, 'Specify the domain to update.'),
+                new InputOption('locales', null, InputOption::VALUE_REQUIRED | InputOption::VALUE_IS_ARRAY, 'Specify the locale to update.', $this->enabledLocales),
+            ])
+            ->setHelp(<<<EOF
+The <info>%command.name%</info> command updates the source tags in XLIFF files with the default locale translation if available.
+
+You can specify directories to update in the command arguments:
+
+  <info>php %command.full_name% path/to/dir path/to/another/dir</info>
+
+To restrict the updates to one or more locales, including the default locale itself, use the <comment>--locales</comment> option:
+
+  <info>php %command.full_name% --locales en --locales fr</info>
+
+You can specify one or more domains to target with the <comment>--domains</comment> option. By default, all available domains for the targeted locales are used.
+
+  <info>php %command.full_name% --domains messages</info>
+
+EOF
+            )
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        $format = $input->getOption('format');
+
+        if (!\array_key_exists($format, self::FORMATS)) {
+            $io->error(sprintf('Unknown format "%s". Available formats are: %s.', $format, implode(', ', array_map(fn ($f) => '"'.$f.'"', array_keys(self::FORMATS)))));
+
+            return self::INVALID;
+        }
```

Comment:
shouldn't we move this to `initialize` method?

Reply:
Not sure about that, I can't return any code from the `initialize` method to indicate the option is invalid, so I would have to throw an exception instead. 

I used the same logic as in [`TranslationUpdateCommand`](https://github.com/symfony/symfony/blob/7.2/src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php#L138).